client/me/billing-history/transactions-table.jsx

```diff
@@ -81,18 +81,26 @@ class TransactionsTable extends React.Component {
 	};
 
 	serviceNameDescription = transaction => {
+		let planName = transaction.plan;
+		// I wish I could use lib/product-values#isJetpackPlan() here, but that always returns `false`
+		// since `transaction.product_slug` uses hyphens (e.g. `jetpack-premium`), whereas `isJetpackPlan()`
+		// internally uses `getPlan()`, which expects underscores (`jetpack_premium`). -- @ockham
+		if ( startsWith( transaction.product_slug, 'jetpack-' ) ) {
+			planName = `Jetpack ${ planName }`;
+		}
+
 		let description;
 		if ( transaction.domain ) {
 			description = (
 				<div>
-					<strong>{ transaction.plan }</strong>
+					<strong>{ planName }</strong>
```

Comment:
I've just noticed a downside of this approach: we display the name in the transactions table with `Jetpack` prepended, however if the user uses the search form and searches "Jetpack Premium" or any other Jetpack plan name, nothing is found. We might need to alter how searching works, or possibly mutate the transaction item object (yeah, doesn't sound optimal 😕).